pkg/api/types.go

```diff
@@ -708,6 +712,14 @@ type ConfigMapVolumeSource struct {
 	Items []KeyToPath `json:"items,omitempty"`
 }
 
+// DockerVolume represents a volume that is backed by a Docker volume plugin. This depends on the plugin being already properly setup.
```

Comment:
Is this really a useful pattern?  It means you can not use volumes that survive pod moves (e.g. the machine crashes and the RC brings the pod up on another machine) because the cleanup will happen.  Who wants this and for whjat kind of volumes?

It seems more useful, especially as a first step, to just use a volume that is managed outside of kubernetes.  Convince me that your model works better?